backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/ClusterFactory.java

```diff
@@ -19,51 +19,260 @@
+
+        public QueryConfigBuilder withDynamicThreshold(PerHostPercentileTracker percentileTracker,
+                                                       double slowQueryLatencyThresholdPercentile) {
+            this.percentileTracker = Optional.of(percentileTracker);
+            this.slowQueryLatencyThresholdPercentile = Optional.of(slowQueryLatencyThresholdPercentile);
+
+            return this;
+        }
+
+        public QueryConfigBuilder withMaxLoggedParameters(int maxLoggedParameters) {
+            this.maxLoggedParameters = Optional.of(maxLoggedParameters);
+
+            return this;
+        }
+
+        public QueryConfigBuilder withMaxParameterValueLength(int maxParameterValueLength) {
+            this.maxParameterValueLength = Optional.of(maxParameterValueLength);
+
+            return this;
+        }
+
+        public QueryConfigBuilder withMaxQueryStringLength(int maxQueryStringLength) {
+            this.maxQueryStringLength = Optional.of(maxQueryStringLength);
+
+            return this;
         }
-        if (refreshSchemaIntervalMillis.isPresent()) {
-            clusterBuilder.withQueryOptions(new QueryOptions().setRefreshSchemaIntervalMillis(refreshSchemaIntervalMillis.get()));
+
+        public QueryLoggerConf build() {
+            Preconditions.checkState(!(constantThreshold.isPresent() && percentileTracker.isPresent()),
+                "You can not use slowQueryLatencyTheresholdMillis and percentileTracker at the same time");
+
+            return new QueryLoggerConf(
+                constantThreshold,
+                percentileTracker,
+                slowQueryLatencyThresholdPercentile,
+                maxLoggedParameters,
+                maxParameterValueLength,
+                maxQueryStringLength
+            );
         }
-        return clusterBuilder.build();
     }
 
-    public static Cluster createClusterForClusterWithoutPassWord(List<CassandraNodeIpAndPort> servers) {
-        return createClusterForClusterWithPassWord(servers, null, null, Optional.empty());
+    public static class QueryLoggerConf {
+        private final Optional<Long> constantThreshold;
+        private final Optional<PerHostPercentileTracker> percentileTracker;
+        private final Optional<Double> slowQueryLatencyThresholdPercentile;
+        private final Optional<Integer> maxLoggedParameters;
+        private final Optional<Integer> maxParameterValueLength;
+        private final Optional<Integer> maxQueryStringLength;
+
+        public static QueryConfigBuilder builder() {
+            return new QueryConfigBuilder();
+        }
+
+        private QueryLoggerConf(
+            Optional<Long> constantThreshold,
+            Optional<PerHostPercentileTracker> percentileTracker,
+            Optional<Double> slowQueryLatencyThresholdPercentile,
+            Optional<Integer> maxLoggedParameters,
+            Optional<Integer> maxParameterValueLength,
+            Optional<Integer> maxQueryStringLength
+        ) {
```

Comment:
This could go on line 122.

Here for sure we never indented like this in James...

```
            Optional<Integer> maxLoggedParameters,
            Optional<Integer> maxParameterValueLength,
            Optional<Integer> maxQueryStringLength) {
```

Is better IMO